libs/model/src/policies/CreateUnverifiedUser.policy.ts

```diff
@@ -0,0 +1,58 @@
+import { Policy } from '@hicommonwealth/core';
+import { events } from '@hicommonwealth/schemas';
+import { Op } from 'sequelize';
+import { ZodUndefined } from 'zod';
+import { models } from '../database';
+
+const inputs = {
+  TokenLocked: events.TokenLocked,
+};
+
+/**
+ * Creates an unverified common user when a new TokenLocked chain event is received and user doesn't exists
+ */
+export function CreateUnverifiedUser(): Policy<typeof inputs, ZodUndefined> {
+  return {
+    inputs,
+    body: {
+      TokenLocked: async ({ payload }) => {
+        const { address } = payload.parsedArgs;
+
+        await models.sequelize.transaction(async (transaction) => {
+          // address already linked to a user
+          const found = await models.Address.findOne({
+            where: { address, user_id: { [Op.ne]: null } },
+            attributes: ['community_id', 'user_id'],
+            transaction,
+          });
+          if (found) return;
+
+          const user = await models.User.create(
+            {
+              email: null,
+              profile: {
+                name: address.substring(0, 8), // TODO: unclaimed name
+                avatar_url: null, // TODO: random avatar
+              },
+            },
+            { transaction },
+          );
+
+          await models.Address.create(
+            {
+              community_id: 'base', // TODO: how to choose a community from payload?
```

Comment:
@dillchen & @sachben91 : do we always select the chain base ethereum community to join these new unverified users? or can we derive a base community from the token event?

Reply:
Hi, yes should, in theory be able to derive the base community. What I image it looks like:

1. Watch for event
2. See Token Event -> see token address associated -> find community associated with the token
3. Add User to Community

Reply:
so, a pre-condition here is that the token must be associated to a community?